src/sinks/gcp/bigquery.rs

```diff
@@ -0,0 +1,193 @@
+use super::{healthcheck_response, GcpAuthConfig, GcpCredentials, Scope};
+use crate::{
+    event::Event,
+    sinks::{
+        util::{
+            http::{BatchedHttpSink, HttpClient, HttpSink},
+            BatchBytesConfig, BoxedRawValue, JsonArrayBuffer, TowerRequestConfig,
+        },
+        Healthcheck, RouterSink, UriParseError,
+    },
+    tls::{TlsOptions, TlsSettings},
+    topology::config::{DataType, SinkConfig, SinkContext, SinkDescription},
+};
+use futures01::Future;
+use http::{Method, Uri};
+use hyper::{Body, Request};
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use snafu::{ResultExt, Snafu};
+use tower::Service;
+
+const NAME: &str = "gcp_bigquery";
+
+#[derive(Debug, Snafu)]
+enum HealthcheckError {
+    #[snafu(display("Configured topic not found"))]
+    TopicNotFound,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+struct BigQueryConfig {
+    project: String,
+    dataset: String,
+    table: String,
+
+    pub emulator_host: Option<String>,
```

Comment:
I am not 100% following why this is called emulator host? Shouldn't this just be called endpoint that you can optionally point to the emulator url?

Reply:
The distinction is that, compared to the AWS sources (which have an endpoint) or other sources (that require a hostname), there is *only one* hostname that is valid for this sink, unless you are using an emulator.

And, in fact, even that is wrong, as there is no emulator for BigQuery that could be used for testing, so I think I should scrap even this vestigial setting.